components/Workspaces/InsightUpgradeModal.tsx

```diff
@@ -26,19 +26,30 @@ export default function InsightUpgradeModal({
           <div className="min-w-[712px] flex flex-col gap-8">
             <DialogCloseButton onClick={onClose} />
             <section className="flex flex-col gap-2">
-              <DialogTitle className="text-xl">This Insight page is over the free Workspace limit</DialogTitle>
-              <p className="text-sm text-slate-500">
-                Your Insight page has{" "}
-                <span className="font-bold">
-                  {overLimit} {variant}
-                </span>{" "}
-                but the free Workspace only allows for{" "}
-                <span className="font-bold">
-                  {variant === "repositories" ? 100 : 10} {variant}
-                </span>{" "}
-                tracked. Don&apos;t worry, your insights won&apos;t be deleted, but if you want to continue using
-                OpenSauced you should upgrade your Workspace to a PRO Workspace.
-              </p>
+              <DialogTitle className="text-xl">
+                {variant !== "workspace"
+                  ? "This Insight page is over the free Workspace limit"
+                  : "Upgrade to a PRO Workspace"}
+              </DialogTitle>
+              {variant !== "workspace" ? (
+                <p className="text-sm text-slate-500">
+                  Your Insight page has{" "}
+                  <span className="font-bold">
+                    {overLimit} {variant}
+                  </span>{" "}
+                  but the free Workspace only allows for{" "}
+                  <span className="font-bold">
+                    {variant === "repositories" ? 100 : 10} {variant}
+                  </span>{" "}
+                  tracked. Don&apos;t worry, your insights won&apos;t be deleted, but if you want to continue using
+                  OpenSauced you should upgrade your Workspace to a PRO Workspace.
```

Comment:
Don't worry, your insights won't be deleted. If you want to continue adding to this Insight Page, you'll need to upgrade your Workspace to a PRO Account.